### PR TITLE
Ap iconnectionspage

### PR DESCRIPTION
**Pull request by Carolien989**
I have one connection multiple times because to see if the blocks are working nicely

### CODE REVIEW

File: Switchboard/public/app/components/connections/ConnectionFactory.js [hunk 1]
```diff
@@ -36,20 +36,11 @@ define(['shared/Registry', 'shared/ResourcesFactory'], function (Registry) {
          * @description
          * Returns all connections
          * 
-         * @returns {connections/ConnectionFactory#Connections} Returns an object that contains all connections
+         * @returns {connections/ConnectionFactory#Connections} Returns an 
+         * object that contains all connections
```
Comment (killje) [inline Switchboard/public/app/components/connections/ConnectionFactory.js]: This no longer returns but a connections/ConnectionFactory#Connections but a promise.
I would write it as Promise.<connections/ConnectionFactory#Connections>

File: Switchboard/public/app/components/connections/OverviewView.html [hunk 2]
```diff
@@ -1,3 +1,12 @@
-<div ng-repeat="connection in connections.connections" ng-click="Page.goToPage('connections/connection/' + connection.connectionId)">
-    <h2>{{connection.name}}</h2>
-</div> 
\ No newline at end of file
+
+<div class="outerbox" ng-repeat="connection in connections" ng-click="connection.click()">
+    <div class="innerbox" ng-class="connection.class" ng-class-odd="'odd'" ng-class-even="'even'" ng-model="myVar" >
+        <h2  class="font1" >{{connection.name}}</h2>
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewView.html]: font1 is very generic, consider replacing this with a more descriptive class name. Both here and in the css

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 3]
```diff
@@ -12,11 +13,32 @@ define(['shared/Registry', 'components/connections/ConnectionFactory'], function
...
+            angular.forEach(connections.connections, function (connection) {
+                connection.click = function () {
+                    Page.goToPage("connections/connection/" + 
+                            connection.connectionId);
+                };
+                connection.class = "connection";
+                $scope.connections.push(connection);
+            });
+
+            var addConnection = {
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: Maybe put this in a function that returns the object

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 4]
```diff
@@ -12,11 +13,32 @@ define(['shared/Registry', 'components/connections/ConnectionFactory'], function
      * @param {connections/ConnectionFactory} ConnectionFactory The connection
      * model that contains the functions to retrieve the information.
      */
-    var OverviewController = function ($scope, ConnectionFactory) {
-        $scope.connections = ConnectionFactory.getConnections();
-    };
+    var OverviewController = function ($scope, Page, ConnectionFactory) {
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: Page is not documented

File: Switchboard/public/app/components/connections/ConnectionFactory.js (hunk 1 above)
Comment (Carolien989) [inline Switchboard/public/app/components/connections/ConnectionFactory.js]: True, i forgot to update/look at all the documentation

**Comment by Carolien989** [comment]
@killje Did everything.  However, while making a function of the var addConnection. It does not regcognize it in the connections array anymore. So i broke it. But i have not  a solution yet other then not making it a function

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 5]
```diff
@@ -12,6 +12,7 @@ define(["shared/Registry", "shared/page/PageFactory",
      * @param {ng/$scope} $scope The scope for this controller
      * @param {connections/ConnectionFactory} ConnectionFactory The connection
      * model that contains the functions to retrieve the information.
+     * @param {Object} Page  Is a PageFactoryObject
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: The order is not the same.
It should not be Object, but page/PageFactory

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 6]
```diff
@@ -25,13 +26,14 @@ define(["shared/Registry", "shared/page/PageFactory",
...
-                name: "create new connection...",
-                class: "apiAddButton",
-                click: function () {
+            var addConnection = function() {
+                this.name = "create new connection...",
+                this.class = "apiAddButton",
+                this.click = function () {
                     alert("TODO");
                 }
             };
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: Not htis kind of function, but:
var addButtonConnections = function(){
    return {
        name: ...,
        class: ...,
        click: ...,
    };
};
$scope.connections.push(addButtonConnection());

Where the function is outside the .then() function

Comment (Carolien989) [inline Switchboard/public/app/components/connections/OverviewController.js]: @killje I get a error message then in this way:
screenshot from 2017-05-10 15-15-45

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 7]
```diff
@@ -19,29 +19,26 @@ define(["shared/Registry", "shared/page/PageFactory",
...
-            $scope.connections.push(addConnection);
-        });
-
-
+        };
+        $scope.connections.push(addButtonConnection());
     };
-    
     OverviewController.$inject = ['$scope', 'page/PageFactory', 'connections/ConnectionFactory'];
-
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: In all other files we have enters around inject and addController. Probably better to stay consistent

File: Switchboard/public/app/components/connections/OverviewController.js [hunk 8]
```diff
@@ -19,29 +19,26 @@ define(["shared/Registry", "shared/page/PageFactory",
...
                     alert("TODO");
                 }
             };
-            
-            $scope.connections.push(addConnection);
-        });
-
-
+        };
+        $scope.connections.push(addButtonConnection());
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewController.js]: This still needs to by inside the then

File: Switchboard/public/app/components/connections/Overview.css [hunk 9]
```diff
@@ -1,9 +1,9 @@
 .innerbox {
-    margin: 10px -1px;
+    margin: -1px;
```
Comment (killje) [inline Switchboard/public/app/components/connections/Overview.css]: should be 0 -1px

File: Switchboard/public/app/components/connections/OverviewView.html [hunk 10]
```diff
@@ -1,11 +1,11 @@
 
 <div class="outerbox" ng-repeat="connection in connections" ng-click="connection.click()">
-    <div class="innerbox" ng-class="connection.class" ng-class-odd="'odd'" ng-class-even="'even'" ng-model="myVar" >
-        <h2  class="font1" >{{connection.name}}</h2>
-        <div class="redX" ng-show="connection.class === 'apiAddButton'"></div>
+    <div class="innerbox" ng-class="connection.class" ng-class-odd="'odd'" ng-class-even="'even'">
+        <h2  class="textlayout" >{{connection.name}}</h2>
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewView.html]: This is not only text, but more, What does this h2 contain?

**Pull request merged**